### DISCUSSION TITLE
[DataGrid] Add function to get items array after sorting and filtering

### DISCUSSION THREAD

**Discussion by toolen** [vmware/clarity#6475] [Ideas]
Hi! I have a pagination table, I need to select a row programmatically, but because of the sorting, I do not know on which page the desired row is located. Then I try to sort the input array and find the desired page with a special function, however, the results of my sorting do not always match the sorting inside the datagrid and the page is incorrect. I tried using ```items._filtered```, but this is a private field that does not always contain up-to-date data. I tried to take the sorting function ```items._sort``` is also not always relevant. It would be great to be able to get a list of all the elements after sorting and filtering from the grid, or for the grid to change the page on its own to highlight the selected row.

**Comment by hippee-lee** [comment]
Getting items on internal state changes has been a repeated request but its a challenge and would require pretty extensive rewriting of the internals due to how they are composed. As we move towards the Core grid element this will not be an issue for applications but it also means we won't be investing in these types of changes for complex components. However, we are not there yet with the core grid so:

1. If it is a server driven datagrid (*ngFor) the array of data given to the grid from the server should be exactly what the server returns and the application should be able to select or mark or augment the data model for the purpose of selecting a row.
2. If the grid is using all data from a local array (*clrDgItems) then using the active sorting and filter values should allow the app to apply filtering and sorting to the list to order it properly outside fo the grid - the catch here is that I don't remember the order sorting/filtering is applied when the datagrid handles it internally.

In both cases If you can supply an [example](https://stackblitz.com/@clr-team) we can take a look at the code and try to give better advice.
